frame/bridge/s2s/issuing/src/lib.rs

```diff
@@ -0,0 +1,374 @@
+//
+// You should have received a copy of the GNU General Public License
+// along with Darwinia. If not, see <https://www.gnu.org/licenses/>.
+
+//! Prototype module for s2s cross chain assets issuing.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub mod weights;
+pub use weights::WeightInfo;
+
+#[cfg(test)]
+mod tests;
+
+// --- crates ---
+use ethereum_types::{H160, H256, U256};
+use sha3::Digest;
+// --- substrate ---
+use frame_support::{
+	ensure,
+	pallet_prelude::*,
+	traits::{Currency, ExistenceRequirement::*, Get},
+	PalletId,
+};
+use frame_system::ensure_signed;
+use sp_runtime::{traits::Convert, DispatchError, SaturatedConversion};
+use sp_std::{str, vec::Vec};
+// --- darwinia ---
+use bp_runtime::{ChainId, Size};
+use darwinia_evm::AddressMapping;
+use darwinia_support::{
+	balance::*,
+	evm::POW_9,
+	s2s::{source_root_converted_id, RelayMessageCaller, ToEthAddress},
+	PalletDigest,
+};
+use dp_asset::{
+	token::{Token, TokenInfo},
+	RecipientAccount,
+};
+use dp_contract::mapping_token_factory::{MappingTokenFactory as mtf, TokenBurnInfo};
+
+const BURN_ACTION: &[u8] = b"burned(address,address,address,address,uint256)";
+
+pub use pallet::*;
+pub type AccountId<T> = <T as frame_system::Config>::AccountId;
+pub type RingBalance<T> = <<T as Config>::RingCurrency as Currency<AccountId<T>>>::Balance;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::config]
+	#[pallet::disable_frame_system_supertrait_check]
+	pub trait Config: dvm_ethereum::Config {
+		#[pallet::constant]
+		type PalletId: Get<PalletId>;
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+		type WeightInfo: WeightInfo;
+		type RingCurrency: LockableCurrency<Self::AccountId, Moment = Self::BlockNumber>;
+
+		type ReceiverAccountId: From<[u8; 32]> + Into<Self::AccountId> + Clone;
+		type BridgedAccountIdConverter: Convert<H256, Self::AccountId>;
+		type BridgedChainId: Get<ChainId>;
+		type ToEthAddressT: ToEthAddress<Self::AccountId>;
+		type OutboundPayload: Parameter + Size;
+		type CallEncoder: EncodeCall<Self::AccountId, Self::OutboundPayload>;
+		type FeeAccount: Get<Option<Self::AccountId>>;
+		type MessageSender: RelayMessageCaller<Self::OutboundPayload, RingBalance<Self>>;
+	}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(PhantomData<T>);
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Handle dispatch call from dispatch precompile contract
+		///
+		/// When user burn their tokens, this handler will receive the event from dispatch
+		/// precompile contract, and relay this event to the target chain to unlock asset.
+		// TODO: update the weight
+		#[pallet::weight(0)]
+		#[frame_support::transactional]
+		pub fn dispatch_handle(origin: OriginFor<T>, input: Vec<u8>) -> DispatchResultWithPostInfo {
```

Comment:
Request better document, function name and weight.

The reason we need this `dispatch_handle` is that the parameters passed from precompile is `input: Vec<u8>`, and we need a handle to parse it?

Reply:
this function is called by dispatch precompile contract to receive the burn event of the mapped token and parse it.
how about changing it to `asset_burn_event_handle`?

Reply:
`asset_burn_event_handle` sound good.